src/main/java/org/example/elements/Animal.java

```diff
@@ -1,30 +1,188 @@
+        child.setGenome(genomeDivide(other));
+
+        return child;
+    }
+
+    public Genome genomeDivide(Animal other){
+        int totalEnergy = this.getEnergy() + other.getEnergy();
+        int thisGenomePartLen = (int) (this.genomeLength*((float) this.getEnergy()/totalEnergy));
+        int otherGenomePartLen = genomeLength - thisGenomePartLen;
+
+        Genome childGenome = new Genome(genomeLength);
+
+        // 0 - left part of stronger Animal
+        // 1 - right part of stronger Animal
+        Random rand = new Random();
+        int side = rand.nextInt() % 2;
+        List<Integer> newGenome = new ArrayList<>();
+
+        if(side == 0 && this.getEnergy() > other.getEnergy()){
+            List<Integer> firstPart = this.genome.getLeftGenomePart(thisGenomePartLen);
+            List<Integer> secondPart = other.genome.getRightGenomePart(thisGenomePartLen);
+            newGenome.addAll(firstPart);
+            newGenome.addAll(secondPart);
+        }
+        else if(side == 1 && this.getEnergy() > other.getEnergy()){
+            List<Integer> firstPart = other.genome.getLeftGenomePart(otherGenomePartLen);
+            List<Integer> secondPart = this.genome.getRightGenomePart(otherGenomePartLen);
+            newGenome.addAll(firstPart);
+            newGenome.addAll(secondPart);
+        } else if (side == 0 && this.getEnergy() <= other.getEnergy()) {
+            List<Integer> firstPart = other.genome.getLeftGenomePart(6);
+            List<Integer> secondPart = this.genome.getRightGenomePart(6);
+            newGenome.addAll(firstPart);
+            newGenome.addAll(secondPart);
+        } else {
+            List<Integer> firstPart = this.genome.getLeftGenomePart(2);
+            List<Integer> secondPart = other.genome.getRightGenomePart(2);
+            newGenome.addAll(firstPart);
+            newGenome.addAll(secondPart);
+        }
+
+        childGenome.setGenomes(newGenome);
+        return childGenome;
+    }
+
+    public void setGenome(Genome genome) {
+        this.genome = genome;
+    }
+
+    public void energyReproduceChange(){
+        initialEnergy -= reproduceEnergy;
+        if(initialEnergy <= 0){
+            isAlive = false;
+        }
+    }
+
+    public void eat(int eatEnergy){
+        initialEnergy += eatEnergy;
+    }
+
+    public void increaseAge(){
+        lifeDays += 1;
     }
 
     public String toString() {
         return direction.toString();
     }
 
+    public MapDirection getDirection() {
+        return direction;
+    }
+
+    public void addObserver(IPositionChangeObserver newObserver) {
+        this.observers.add(newObserver);
+    }
+
+    public void removeObserver(IPositionChangeObserver observerToRemove) {
+        this.observers.remove(observerToRemove);
+    }
+
     @Override
     public String getPath() {
-        return super.getPath();
+        return switch (getDirection()) {
```

Comment:
Jeżeli damy radę to lepiej będzie robić coś w stylu `getClass().getResources()` itd.